File: legacy-php/Player.php
<?php
class Player
{
	private $_name = '';
	private $_number = '';
	private $_offense = 0;
	private $_defense = 0;
	private $_stamina = 0;

	public function __construct($defaultStrength = null, $position = null)
	{
		$this->_getRandom($defaultStrength, $position);
	}

	private function _getRandom($defaultStrength = null, $position = null)
	{
		$this->_name = NameGenerator::getRandom();

		if ($position == 'G') {
			$this->_number = 1;
			$this->_offense = array_rand(array(0, 0, 0, 0, 0, 0, 2, 3, 3, 7));
			$this->_defense = ($defaultStrength != null ? $defaultStrength : rand(8, 10));
			$this->_stamina = ($defaultStrength != null ? $defaultStrength : rand(0, 10));
		} else {
			$this->_number = rand(1, 99);
			$this->_offense = ($defaultStrength != null ? $defaultStrength : rand(0, 10));
			$this->_defense = ($defaultStrength != null ? $defaultStrength : rand(0, 10));
			$this->_stamina = ($defaultStrength != null ? $defaultStrength : rand(0, 10));
		}
	}

	public function getOffense()
	{
		return $this->_offense;
	}

	public function getDefense()
	{
		return $this->_defense;
	}

	public function getStamina()
	{
		return $this->_stamina;
	}
}

File: app/models/player.rb
class Player < ActiveRecord::Base
  belongs_to :real_team
  
  has_many :teams, through: :contracts
  
  validates :name, presence: true
  validates :number, presence: true
  validates :position, inclusion: {in: %w(G D M O)}
  validates :real_team_id, presence: true
  
  validates :offense, numericality: {greater_than_or_equal_to: 0, less_than_or_equal_to: 10}
  validates :defense, numericality: {greater_than_or_equal_to: 0, less_than_or_equal_to: 10}
  validates :stamina, numericality: {greater_than_or_equal_to: 0, less_than_or_equal_to: 10}
end